deployment/terraform/cloud-config/ecs-batch-user-data.yml

```diff
@@ -3,19 +3,22 @@
 packages:
     - awslogs
 
-fs_setup:
-  - label: nvme0n1
-    filesystem: ext4
-    extra_opts: ["-E", "nodiscard"]
-    device: /dev/nvme0n1
-    partition: auto
-
-mounts:
-  - [ /dev/nvme0n1, null ]
-  - [ /dev/nvme0n1, "/media/nvme0n1", "ext4", "defaults,nobootwait,discard", "0", "2" ]
-
 runcmd:
   - curl -o /etc/papertrail-bundle.pem https://papertrailapp.com/tools/papertrail-bundle.pem
+  # Manually mount i3 family ephemeral storage, see:
+  # https://bugs.launchpad.net/cloud-init/+bug/1672833
+  # o - Create empty dos partition table
+  # n - Add new partition
+  # p - Primary partition (In add new prompt)
+  # 1 - Partition number (In add new prompt)
+  # empty newline - First sector, accepts default: 1
+  # empty newline - Last sector, accepts default: end
+  # w - Write changes
+  - echo -e "o\nn\np\n1\n\n\nw" | fdisk /dev/nvme0n1
+  - mkfs.ext4 /dev/nvme0n1p1
+  - mkdir -p /media/nvme0n1p1
+  - echo -e "/dev/nvme0n1p1\t/media/nvme0n1p1\text4\tdefaults,discard\t0\t0" >> /etc/fstab
```

Comment:
For similar reasons:

```yaml
- echo -e "/dev/nvme0n1p1\t/media/nvme0n1p1\text4\tdefaults,nofail,discard\t0\t2" >> /etc/fstab
```

- `nofail`: Bomb boot sequence if drive fails to mount
- Set `fs_passno` to 2 (the last column)

> This field is used by fsck(8) to determine the order in which filesystem checks are done at boot time.  The root filesystem should be specified with a fs_passno of 1.  Other filesystems should have a fs_passno of 2.